src/cpu/aarch64/jit_sve_512_x8s8s32x_conv_kernel.cpp

```diff
@@ -0,0 +1,1642 @@
+            jcp.t_pad, jcp.oh, jcp.ih, jcp.stride_h, ext_kh);
+    jcp.back_pad = calculate_end_padding(
+            jcp.f_pad, jcp.od, jcp.id, jcp.stride_d, ext_kd);
+    bool kernel_outside_src = false || ext_kw <= jcp.l_pad
+            || ext_kw <= jcp.r_pad || ext_kh <= jcp.t_pad || ext_kh <= jcp.b_pad
+            || ext_kd <= jcp.f_pad || ext_kd <= jcp.back_pad;
+    if (kernel_outside_src) return status::unimplemented;
+
+    jcp.signed_input = (src_d.data_type() == data_type::s8) ? true : false;
+    jcp.need_saturation = utils::one_of(dst_d.data_type(), u8, s8, s32);
+    jcp.is_depthwise = true && with_groups && everyone_is(1, jcp.ic, jcp.oc);
+
+    if (jcp.is_depthwise && is_3d)
+        // NOTE: 3D depthwise is not currently supported here.
+        return status::unimplemented;
+
+    if (jcp.is_depthwise) {
+        jcp.ch_block = 16;
+        jcp.ic_block = 1;
+        jcp.oc_block = 1;
+    } else {
+        jcp.ch_block = 1;
+        jcp.ic_block = 16;
+        jcp.oc_block = 16;
+
+        if (jcp.ngroups == 1) {
+            /* For non grouped convolutions, pad channels by 16 if needed */
+            jcp.oc = rnd_up(jcp.oc, jcp.oc_block);
+            jcp.ic = rnd_up(jcp.ic, jcp.ic_block);
+        } else if (jcp.ngroups != 1
+                && ((jcp.ic % jcp.ic_block != 0)
+                        || (jcp.oc % jcp.oc_block != 0))) {
+            /* For grouped convolutions, oneDNN doesn't support padding.
+               When channels per group is not multiple of 16:
+                - Use Ymm when channels per group is multiple of 8,
+                - Use Xmm when channels per group is multiple of 4,
+                - Otherwise return unimplemented. */
+            jcp.ic_block = (jcp.ic % 8 == 0) && (jcp.oc % 8 == 0) ? 8 : 4;
+            jcp.oc_block = jcp.ic_block;
+        }
+        if (jcp.ic % jcp.ic_block != 0 || jcp.oc % jcp.oc_block != 0)
+            return status::unimplemented;
+    }
+
+    if (!post_ops_ok(jcp, attr)) return status::unimplemented;
+
+    const auto &p = attr.post_ops_;
+
+    const auto zp = attr.zero_points_;
+    jcp.dst_zero_point = !zp.has_default_values(DNNL_ARG_DST);
+    jcp.src_zero_point = !zp.has_default_values(DNNL_ARG_SRC);
+    if (jcp.dst_zero_point || jcp.src_zero_point) return status::unimplemented;
+
+    jcp.ver = ver_sve;
+    jcp.is_fast_depthwise = true && jcp.is_depthwise
+            && jcp.ngroups % jcp.ch_block == 0; /* groups not multiple of
+    ch_block (= 16) would require byte masking for load from src */
+
+    jcp.is_resrc_depthwise = jcp.is_depthwise && jcp.stride_w < jcp.kw
+            && jcp.kw < 4 && jcp.dilate_w == 0;
+    if (jcp.is_depthwise) {
+        jcp.max_regs_ur = 31 - jcp.is_fast_depthwise - !jcp.is_resrc_depthwise
+                - !jcp.signed_input
+                - (!jcp.signed_input || jcp.need_saturation); // both alias
+    } else {
+        // jcp.max_regs_ur = jcp.ver == ver_vnni ? 31 : 28;
+        jcp.max_regs_ur = 31;
```

Comment:
[Random place]
As I understand, `sdot` is counterpart of the `vnni` instruction in `x64` architecture. I was wondering if it also has the limitation about being applicable only for mixed unsigned and signed input [especially because downstairs, there is code related with compensation aspect]

Moreover, maybe we also need to update the documentation [here!](https://oneapi-src.github.io/oneDNN/dev_guide_int8_computations.html)

Reply:
Since the operands of the 8-bit integer dot4 product instruction (```sdot```) of aarch64 are src1_dtype=s8, src2_dtype=s8 and dst_dtype=s32, the int8 convolution using ```sdot``` of aarch64 computes for mixed unsigned and signed inputs as follows:

```
Ys32 = Xu8⋅Ws8 = X's8⋅Ws8 + 128⋅Ws8
X's8 = Xu8 - 128
```

Reply:
Thanks for the clarifying details!